packages/cw-croncat-core/src/types.rs

```diff
@@ -111,17 +111,39 @@ impl BoundaryValidated {
     ) -> Result<Self, CoreError> {
         if let Some(boundary) = boundary {
             match (interval, boundary) {
-                (Interval::Cron(_), Boundary::Time { start, end }) => Ok(Self {
-                    start: start.map(|start| start.nanos()),
-                    end: end.map(|end| end.nanos()),
-                }),
+                (Interval::Cron(_), Boundary::Time { start, end }) => match (start, end) {
+                    (Some(s), Some(e)) => {
+                        if s.nanos() >= e.nanos() {
+                            return Err(CoreError::InvalidBoundary {});
+                        }
+                        Ok(Self {
+                            start: start.map(|start| start.nanos()),
+                            end: end.map(|end| end.nanos()),
```

Comment:
```suggestion
                            start: Some(s.nanos()),
                            end: Some(e.nanos()),
```
No need to do second patternmatch here!